### ISSUE TITLE
Kill to the clipboard is frustrating

### ISSUE THREAD

**Issue by timbertson** [fish-shell/fish-shell#772]
I often find myself doing something of the form:

    $ some long command with <URL>

If I'm going through it with a bunch of URLs, I'll copy the next URL to my clipboard, press &lt;up&gt;, and then &lt;alt-bksp&gt; to remove the last argument. But then when I paste, I'm pasting the URL I just removed because fish pushed it to my clipboard!

Personally, I'd like fish to not put anything to the clipboard (or even the selection buffer) when it deletes stuff. But clearly this was added for a reason, should it be configurable? Is there any way I can hack out fish's ability to kill to the clipboard with a ./configure flag perhaps?

**Comment by zanchey** [comment]
Perhaps the default should be "do nothing" and document setting FISH_CLIPBOARD_CMD to `xsel` if you want to keep the existing behaviour?

**Comment by maxfl** [comment]
It's a bit offtopic here, but would it be better having 'fish_clipboard' function instead of environment variable? The function is consistent with other fish functions: fish_title, fish_prompt, fish_right_prompt.
It also could be handled via events. Having variables-hook-functions and plain-hook-functions in the same time seems to be very inelegant.

**Comment by zanchey** [comment]
Sounds good to me.

**Comment by nh2** [comment]
I don't have a particular opinion about this, but can anybody explain me why I cannot `Ctrl+w` (deleting the last word, works) and then `Ctrl+y` (yanking that work, doesn't work)? It seems to be related to this issue.

**Comment by brunetton** [comment]
+1 for @maxfl proposal
And thanks to @gfxmonk for workaround

**Comment by gechr** [comment]
I agree this is quite frustrating and +1 @maxfl's proposal as well - would be the best of both worlds.

**Comment by simonbcn** [comment]
:+1:

**Comment by krader1961** [comment]
Closing this as a dup of issue #2894. I'm going to kill this "feature". If anyone thinks a `fish_clipboard` function (akin to `fish_title`) is a good idea I ask that a new enhancement request be opened with a suitable argument for its addition. Please provide scenarios illustrating why this would be useful and how such a function might be used in key bindings.

**Comment by foxish** [comment]
The best way I found to fix this in v2.0.0 was to do the following:

```shell
remove xsel (apt-get remove xsel if applicable)
sudo cp /bin/true /usr/bin/xsel
```

This removes kill-to-clipboard for now. Thanks to one of the comments above for the xsel solution.
